rectools/model_selection/last_n_split.py

```diff
@@ -103,8 +102,10 @@ def _split_without_filter(
         df = interactions.df
         idx = pd.RangeIndex(0, len(df))
 
-        # last event - rank=1
-        inv_ranks = df.groupby(Columns.User)[Columns.Datetime].rank(method="first", ascending=False)
+        # Here we guarantee that last appeared interaction in df will have lowest rank when datetime is not unique
+        time_order = df.groupby(Columns.User)[Columns.Datetime].rank(method="first", ascending=True).astype(int)
+        n_interactions = df.groupby(Columns.User).transform("size").astype(int)
```

Comment:
Let's also avoid double groupby here for faster computations:
```suggestion
    grouped = df.groupby(Columns.User)
    time_order = grouped[Columns.Datetime].rank(method="first", ascending=True).astype(int)
    n_interactions = grouped[Columns.User].transform('size').astype(int)
```